### ISSUE TITLE
Rename PMComplex to PMComplexNumber

### ISSUE THREAD

**Issue by olekscode** [PolyMathOrg/PolyMath#191]

**Comment by SergeStinckwich** [comment]
I'm not sure we should go in that direction. For example, there is a ``Float`` or ``Integer`` not ``FloatNumber`` or ``IntegerNumber``.

**Comment by nicolas-cellier-aka-nice** [comment]
The main grief was with vocable `isComplex` which is way too broad and subject to false polymorphism.
A lot of objects can be complex, without being complex numbers.
In VW, there was at least a case of such false polymorphism (in ProgramNode).
Same in Squeak (ParseNode) and (ComplexBorder).

So the initial proposal was to rename `isComplex` to `isComplexNumber`.
I think that the class renaming idea originated from this pattern: `Integer >> #isInteger`, `Float >> #isFloat`, etc...
But with the PM prefix, maybe the renaming is moot.
Unless we rename `isComplex` -> `isPMComplex`...
Beware: `isPMComplex` might reduce possibilities of extensions/intercompatibility because on the contrary to `isComplex`, it could be seen as much too restrictive. Some other species might quake like a complex...

**Comment by olekscode** [comment]
Well, to be honest, i also think that in a general scenario, `Integer` and `Float` should be `IntegerNumber` and `FloatNumber`.

But because `int` and `float` are super-standard terms in every programmer's vocabulary, as an exception, we can keep those class names simple.

But `Complex` is a different story. It is an adjective (class names should be nouns!) that is often used in various contexts: complex view, complex scenario, etc. This can confuse many programmers who read the code, especially those who are not mathematicians.

In other words, this:

```Smalltalk
x := PMComplex new.
x isComplex.
x asComplex.
```

seems to me much less descriptive than this:

```Smalltalk
x := PMComplexNumber new.
x isComplexNumber.
x asComplexNumber.
```

**Comment by hemalvarambhia** [comment]
I would like weigh in, if I may. I'm wondering about this from a domain driven  perspective. I think it should be PMComplexNumber, and Integer remain the same. This rename makes sense to me with my background, as we already referred to 'integers' and 'complex numbers' in conversation. We should refactor to what is proposed here.

There is a wider question of the project appealing to Domain Driven Design. I think it should. Is there any reason we shouldn't adopt Domain Driven Design (DDD)? At the very least, let's have the code be as close as possible to the domain language of mathematics. 

@olekscode you work with Cyrille Martraire so it would be worth getting his thoughts because he is an expert DDD practitioner.

As, Eric Evans, who invented DDD, and was inspired to do so from his Smalltalk background. Perhaps we can follow that lead.

**Comment by hemalvarambhia** [comment]
From the wiki:

The set {\displaystyle \mathbb {C} }￼ of complex numbers is a field.[47] Briefly, this means that the following facts hold: first, any two complex numbers can be added and multiplied to yield another complex number. Second, for any complex number z, its additive inverse –z is also a complex number; and third, every nonzero complex number has a reciprocal complex number. Moreover, these operations satisfy a number of laws, for example the law of commutativity of addition and multiplication for any two complex numbers z1 and z2:
